nntrainer/models/neuralnet.cpp

```diff
@@ -430,10 +430,14 @@ void NeuralNetwork::readModel() {
     (*iter)->read(model_file);
   }
 
-  checkedRead(model_file, (char *)&tmp.epoch_idx, sizeof(epoch_idx),
-              "[NeuralNetwork::readModel] failed to read epoch_idx");
-  checkedRead(model_file, (char *)&tmp.iter, sizeof(iter),
-              "[NeuralNetwork::readModel] failed to read iteration");
+  try {
+    checkedRead(model_file, (char *)&tmp.epoch_idx, sizeof(epoch_idx),
+                "[NeuralNetwork::readModel] failed to read epoch_idx");
+    checkedRead(model_file, (char *)&tmp.iter, sizeof(iter),
+                "[NeuralNetwork::readModel] failed to read iteration");
+  } catch (...) {
+    std::cerr << "failed to read epoch idx, proceeding with default index\n";
```

Comment:
please add a comment here that this failure is allowed at the end of file read. If this has failed, no other read operation should be performed on this file.
Or maybe move model_file.close() inside try and catch both to emphasize on it.

Reply:
Good idea, added :)